src/converters/app-converter.ts

```diff
@@ -142,12 +142,36 @@ export class StackConverter extends ArtifactConverter {
                 const options = this.processOptions(cfn, parent);
 
                 const mapped = this.mapResource(n.logicalId, cfn.Type, props, options);
-                mapped.forEach((m) => {
-                    const resource = pulumi.Resource.isInstance(m) ? m : m.resource;
-                    const attributes = pulumi.Resource.isInstance(m) ? undefined : m.attributes;
-                    this.resources.set(n.logicalId!, { resource, attributes, resourceType: cfn.Type });
-                    this.constructs.set(n.construct, resource);
-                });
+                const mainResource: ResourceAttributeMapping | undefined = Array.isArray(mapped)
```

Comment:
This deserves a separate quick PR? Some change was necessary to enable additional tests?